### ISSUE TITLE
DataViewer: Failed Pandas dependency check (false negative) causes error message

### ISSUE THREAD

**Issue by galloppinggryphon** [microsoft/vscode-jupyter#15213]
## tl;dr
DataViewer asks for Pandas to be installed every time the DataViewer is opened. It seems to be caused by a brittle Pandas check performed by Jupyter that fails if `import pandas` displays a warning (instead of nothing). In my case, Pandas displayed a warning about a missing package that would become a future dependency (`pyarrow`). Installing pyarrow fixed the problem.

## Environment data

-   VS Code version: `1.86.2`
-   Jupyter Extension version (available under the Extensions sidebar): `2024.1.1`
-   Python Extension version (available under the Extensions sidebar): `2024.0.1`
-   OS (Windows | Mac | Linux distro) and version: `Win11 22.H2`
-   Python and/or Anaconda version: `Python 3.12.1`
-   Type of virtual environment used (N/A | venv | virtualenv | conda | ...): `n/a`
-   Jupyter server running: `n/a`

## Expected behaviour

The DataViewer in Jupyter notebooks should pop open without error if a recent version of Pandas is installed.

## Actual behaviour

When opening the DataViewer from the `Variables` section of the `Jupyter` panel, VSCode produces an error dialogue:
`Python package "pandas" version 0.20.0 is required for viewing data. [Install] | [Cancel]`

After pressing `Install`, the DataViewer opens normally.

The error message is shown every time the DataViewer is opened.

XXX

## Steps to reproduce:

1. Install Pandas `v2.2.0` globally

1. Create a minimal DataFrame in a blank Jupyter notebook cell and execute:

```
import pandas as pd
df = pd.DataFrame([1,2,3])
```

2. Open the `Jupyter` panel and click the popup icon next to the `df` variable.

![image](https://github.com/microsoft/vscode-jupyter/assets/26875847/9e598dc4-d5e2-4694-8f98-3f9b2774a32c)

3. Behold the resulting error

![image](https://github.com/microsoft/vscode-jupyter/assets/26875847/4a118377-1eb8-4922-88c4-45f263098b47)

## Apparent cause and how to fix

When imported, Pandas displays a warning:
```
Failed to get version of Pandas to use the Data Viewer. <string>:1: DeprecationWarning: 
Pyarrow will become a required dependency of pandas in the next major release of pandas (pandas 3.0),
(to allow more performant data types, such as the Arrow string type, and better interoperability with other libraries)
but was not found to be installed on your system.
```

After installing the missing `pyarrow` (`pip install pyarrow`), the DataViewer pops open without error.

From the log (included below), it seems the presence of this warning causes the Jupyter to conclude Pandas is not installed:
`~\AppData\Local\Programs\Python\Python312\python.exe -c "import pandas;print(pandas.__version__);print("5dc3a68c-e34e-4080-9c3e-2a532b2ccb4d")"`

## Logs

<details>

<summary>Output for <code>Jupyter</code> in the <code>Output</code> panel (<code>View</code>→<code>Output</code>, change the drop-down the upper-right of the <code>Output</code> panel to <code>Jupyter</code>)
</summary>

<p>

```
18:05:10.044 [info] Handle Execution of Cells 0 for ~\Development\Python\Test\test_notebook.ipynb
18:05:10.049 [info] Kernel acknowledged execution of cell 0 @ 1708470310048
18:05:10.054 [info] End cell 0 execution after 0.006s, completed @ 1708470310054, started @ 1708470310048
18:05:20.230 [warn] Failed to get activated env vars for ~\AppData\Local\Programs\Python\Python312\python.exe in 107ms
18:05:20.238 [info] Process Execution: ~\AppData\Local\Programs\Python\Python312\python.exe -c "import pandas;print(pandas.__version__);print("5dc3a68c-e34e-4080-9c3e-2a532b2ccb4d")"
18:05:20.863 [warn] Failed to get version of Pandas to use the Data Viewer. <string>:1: DeprecationWarning: 
Pyarrow will become a required dependency of pandas in the next major release of pandas (pandas 3.0),
(to allow more performant data types, such as the Arrow string type, and better interoperability with other libraries)
but was not found to be installed on your system.
If this would cause problems for you,
please provide us feedback at https://github.com/pandas-dev/pandas/issues/54466
        

18:05:25.048 [error] [Error: Python package 'pandas' version 0.20.0 (or above) is required for viewing data.
	at $0.promptInstall (~\.vscode\extensions\ms-toolsai.jupyter-2024.1.1-win32-x64\dist\extension.node.js:362:50277)]
```

**Note: this is after pressing cancel instead of install.**

</p>
</details>

**Comment by amunger** [comment]
duplicate of https://github.com/microsoft/vscode-jupyter/issues/15128. should be fixed in the latest release.
